Python: rework documentation (#5357)
github/codeql: docs/codeql/codeql-language-guides/codeql-for-python.rst

   basic-query-for-python-code
   codeql-library-for-python
   analyzing-data-flow-in-python
   functions-in-python
   expressions-and-statements-in-python
   analyzing-control-flow-in-python

-  :doc:`Basic query for Python code <basic-query-for-python-code>`: Learn to write and run a simple CodeQL query using LGTM.

-  :doc:`CodeQL library for Python <codeql-library-for-python>`: When you need to analyze a Python program, you can make use of the large collection of classes in the CodeQL library for Python.

> -  :doc:`Analyzing data flow in Python <analyzing-data-flow-in-python>`: You can use CodeQL to track the flow of data through a Python program. Tracking user-controlled, or tainted, data is a key technique for security researchers.

Reviewer: ```suggestion
-  :doc:`Analyzing data flow in Python <analyzing-data-flow-in-python>`: You can use CodeQL to track the flow of data through a Python program to places where the data is used.
```

(Copied from the new page's intro)
